app/src/gpio_key_wakeup_trigger.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/init.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/pm.h>
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#define DT_DRV_COMPAT zmk_gpio_key_wakeup_trigger
+
+struct gpio_key_wakeup_trigger_config {
+    struct gpio_dt_spec trigger;
+    size_t extra_gpios_count;
+    struct gpio_dt_spec extra_gpios[];
+};
+
+static int zmk_gpio_key_wakeup_trigger_init(const struct device *dev) {
+#if IS_ENABLED(CONFIG_PM_DEVICE)
+    pm_device_init_suspended(dev);
+    pm_device_wakeup_enable(dev, true);
+#endif
+
+    return 0;
+}
+
+#if IS_ENABLED(CONFIG_PM_DEVICE)
+
+static int gpio_key_wakeup_trigger_pm_resume(const struct device *dev) {
+    const struct gpio_key_wakeup_trigger_config *config = dev->config;
+
+    int ret = gpio_pin_interrupt_configure_dt(&config->trigger, GPIO_INT_LEVEL_ACTIVE);
```

Comment:
The logic for resume/suspend feels backwards to me, because I was thinking of this in terms of the power state of the system until I read pm.c. This might warrant to comment somewhere to explain that this device is normally suspended and then gets resumed just prior to going into soft off.

Reply:
Added a comment there